hotel/report/hotel_report.py

```diff
@@ -1,57 +1,80 @@
                       ('checkout_date', '<=', date_end)]
         tids = folio_obj.search(act_domain)
         for data in tids:
-            checkin = data.checkin_date.\
-                strftime(DEFAULT_SERVER_DATETIME_FORMAT)
-            checkout = data.checkout_date.\
-                strftime(DEFAULT_SERVER_DATETIME_FORMAT)
+            checkin = self.format_date(data.checkin_date, user_lang, to_zone)
+            checkout = self.format_date(data.checkout_date, user_lang, to_zone)
             data_folio.append({
                 'name': data.name,
                 'partner': data.partner_id.name,
-                'checkin': parser.parse(checkin),
-                'checkout': parser.parse(checkout),
+                'checkin': checkin,
+                'checkout': checkout,
                 'amount': data.amount_total
             })
             total_amount += data.amount_total
         data_folio.append({'total_amount': total_amount})
         return data_folio
 
+    def format_date(self, date, user_lang, to_zone):
+        if isinstance(date, str):
+            new_date = datetime.strptime(date, '%Y-%m-%d %H:%M:%S')
+        else:
+            new_date = date
+        timezone_conv = pytz.timezone(to_zone)
+        timezone_utc = pytz.timezone('UTC')
+        new_date = timezone_utc.localize(new_date)
+        new_date = new_date.astimezone(timezone_conv)
+        date_converted = new_date.strftime(user_lang.date_format+" "+user_lang.time_format)
+        return date_converted
+
     @api.model
     def _get_report_values(self, docids, data):
         self.model = self.env.context.get('active_model')
         if data is None:
             data = {}
         if not docids:
             docids = data['form'].get('docids')
-        folio_profile = self.env['hotel.folio'].browse(docids)
-        date_start = data['form'].get('date_start', fields.Date.today())
-        date_end = data['form'].get(
-            'date_end', str(datetime.now() + relativedelta(
-                months=+1, day=1, days=-1))[:10])
+        context = self._context
+        current_uid = context.get('uid')
+        user = self.env['res.users'].browse(current_uid)
+        user_lang = self.env['res.lang'].search([('code', '=', user.lang)])[0]
+        if self._context.get('tz'):
+            to_zone = self._context.get('tz')
+        else:
+            to_zone = 'UTC'
+        rm_act = self.with_context(data['form'].get('used_context', {}))
+        folio_profile = self.env['hotel.reservation'].browse(docids)
+        date_start = data['form'].get('date_start', str(datetime.now())[:10])
+        date_end = data['form'].get('date_end', str(datetime.now() +
+                                    relativedelta(months=+1,
+                                                  day=1, days=-1))[:10])
+        date_start_formatted = rm_act.format_date(date_start, user_lang, to_zone)
+        date_end_formatted = rm_act.format_date(date_end, user_lang, to_zone)
```

Comment:
Would be very helpful if you add comments per section you write, would help developers.